internal/system/agent/services.go

```diff
@@ -81,54 +86,94 @@ func InvokeOperation(action string, services []string) bool {
+	configuration := make(map[string]interface{})
 
 	// Loop through requested actions, along with respectively-supplied parameters.
 	for _, service := range services {
 
-		c.Configuration[service] = ""
+		configuration[service] = ""
 
 		if !isKnownServiceKey(service) {
 			logs.LoggingClient.Warn("unknown service found getting configuration", "service name", service)
 		}
 
 		responseJSON, err := clients[service].FetchConfiguration(ctx)
 		if err != nil {
-			c.Configuration[service] = fmt.Sprintf("%s get config error: %s", service, err.Error())
+			configuration[service] = fmt.Sprintf("%s get config error: %s", service, err.Error())
 			logs.LoggingClient.Error("error retrieving configuration", "service name", service, "error message", err.Error())
 		} else {
-			c.Configuration[service] = ProcessResponse(responseJSON)
+			configuration[service] = ProcessResponse(responseJSON)
 		}
 	}
-	return c, nil
+	return configuration, nil
 }
 
-func getMetrics(services []string, ctx context.Context) (MetricsRespMap, error) {
+func getMetrics(services []string, ctx context.Context) (map[string]interface{}, error) {
 
-	m := MetricsRespMap{}
-	m.Metrics = map[string]interface{}{}
+	metrics := make(map[string]interface{})
 
 	// Loop through requested actions, along with respectively-supplied parameters.
 	for _, service := range services {
 
-		m.Metrics[service] = ""
+		metrics[service] = ""
 
 		if !isKnownServiceKey(service) {
 			logs.LoggingClient.Warn("unknown service found getting metrics", "service name", service)
 		}
 
 		responseJSON, err := clients[service].FetchMetrics(ctx)
 		if err != nil {
-			m.Metrics[service] = fmt.Sprintf("%s get metrics error: %s", service, err.Error())
+			metrics[service] = fmt.Sprintf("%s get metrics error: %s", service, err.Error())
 			logs.LoggingClient.Error("error retrieving metrics", "service name", service, "error message", err.Error())
 		} else {
-			m.Metrics[service] = ProcessResponse(responseJSON)
+			metrics[service] = ProcessResponse(responseJSON)
 		}
 	}
-	return m, nil
+	return metrics, nil
+}
+
+func getHealth(services []string) (map[string]interface{}, error) {
+
+	health := make(map[string]interface{})
+
+	config := registry.Config{
+		Host:            Configuration.Registry.Host,
+		Port:            Configuration.Registry.Port,
+		Type:            Configuration.Registry.Type,
+		ServiceKey:      internal.SystemManagementAgentServiceKey,
+		ServiceHost:     Configuration.Service.Host,
+		ServicePort:     Configuration.Service.Port,
+		ServiceProtocol: Configuration.Service.Protocol,
+		CheckInterval:   Configuration.Service.CheckInterval,
+		CheckRoute:      clientsMod.ApiPingRoute,
+		Stem:            internal.ConfigRegistryStem,
+	}
+	client, err := factory.NewRegistryClient(config)
+	if err != nil {
+		logs.LoggingClient.Error("could not create registry client", "errMsg", err.Error())
+		return health, err
+	}
+
+	for _, service := range services {
+
+		health[service] = ""
```

Comment:
why is this initialized here? It's always set to something else below in all control flows